### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by AnyoneClown**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,4 +2,61 @@ class Cargo:
...
-# write your code here
+
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
+        self.name = name
+        self.weight = weight
+        if coords is None:
+            self.coords = [0, 0]
+        else:
+            self.coords = coords
```
Comment (vsmutok) [inline app/main.py:13]: use `or` here

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 2]
```diff
@@ -2,4 +2,56 @@ class Cargo:
...
+            weight: int,
+            max_load_weight: int,
+            current_load: Cargo | None,
+            coords: list = None
+    ) -> None:
+        super().__init__(name, weight, coords)
+        self.max_load_weight = max_load_weight
+        self.current_load = current_load
+
+    def hook_load(self, other: Cargo) -> None:
```
Comment (viktoria-rybenchuk) [inline app/main.py:52]: use descriptive parameter name instead of `other`

File: app/main.py [hunk 3]
```diff
@@ -2,4 +2,56 @@ class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(self, name: str, weight: int, coords: list = None) -> None:
```
Comment (viktoria-rybenchuk) [inline app/main.py:7]: specify type annotations for `coords`

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**